src/main/java/vazkii/botania/common/block/tile/TileAlfPortal.java

```diff
@@ -137,7 +137,17 @@ public void update() {
 							continue;
 
 						ItemStack stack = item.getEntityItem();
-						if((!(stack.getItem() instanceof IElvenItem) || !((IElvenItem) stack.getItem()).isElvenItem(stack)) && !item.getEntityData().hasKey(TAG_PORTAL_FLAG)) {
+						boolean consume;
+						if(item.getEntityData().hasKey(TAG_PORTAL_FLAG)) {
+							consume = false;
+						} else if(stack.getItem() instanceof ItemLexicon) {
+							consume = true;
```

Comment:
ItemLexicon is an `IElvenItem` itself, is there a way you can fold this branch of this if into the one below this comment?